source/20_spec/config_spec.rst

```diff
@@ -0,0 +1,154 @@
+Specifications of DI-engine Config
+==================================
+
+To ensure the ease of use, readability, and extensibility of the config，config submitted by the developers should comply with the following specifications.
+
+Config of DI-engine includes two divisions：main_config and create_config.
+
+Example Link
+--------------
+
+Example of Deep Q-Network （DQN）：
+
+https://github.com/opendilab/DI-engine/blob/main/dizoo/atari/config/serial/pong/pong_dqn_config.py
+
+Example of SQIL with model or data:
+
+https://github.com/opendilab/DI-engine/blob/main/dizoo/atari/config/serial/pong/pong_sqil_config.py
+
+Details of the specification
+------------------------------
+
+Specification of Grammar
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+-  config should satisfy flake8 python syntax checking and yapf formatting.
+
+Specification of naming 
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+-  file name： config.py,related variable name：main_config and create_config
+
+   -  Uniformly named after <env_name>\_<algo_name>\_config.py
+      .The name of the file and related variable names in the file do not need to add the default field. For instance file name hopper_onppo_default_config.py should be changed into hopper_onppo_config.py。
+
+   -  Similarly
+      For ICM algorithm,the general algorithm is the module proposed in the paper combined with a baseline algorithm,its corresponding config name should be named as <env_name>\_<module_name>\_<baseline_name>\_config.py
+      ,such as cartpole_icm_offppo_config.py
+
+   -  If the algorithm has verious versions of on-policy and off-policy ,please unify the name of config.py file and related varible names in the file,and use of onppo/offppo to distinguish on-policy and off-policy versions of the algorithm. For example,for the config of the PPO algorithm,
+      hopper_ppo_config.py should be changed to hopper_onppo_config.py。
+
+-  exp_name field
+
+   -  main_config must include exp_name filed
+
+   -  The naming convention is environemnt+algorithm+seed,such as \ ``qbert_sqil_seed0``
+
+-  Name of the file path
+
+   -  See the sqil example, commented accordingly.If multiple models need to be loaded, the model path (model_path) variable is named as follows：prefix1_model_path,prefix2_model_path,...,
```

Comment:
Please refer to the [sqil ](https://github.com/opendilab/DI-engine/blob/main/dizoo/atari/config/serial/pong/pong_sqil_config.py) example, commented accordingly. If multiple models need to be loaded, the model path variable is named as follows：`prefix1_model_path`, `prefix2_model_path`, .... The varibles of data_path are named in same way.